src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -9,21 +11,31 @@
 import seedu.address.logic.commands.CommandResult;
 import seedu.address.logic.parser.Parser;
 import seedu.address.model.Model;
+import seedu.address.model.TaskBook;
 import seedu.address.model.task.DeadlineTask;
 import seedu.address.model.task.EventTask;
 import seedu.address.model.task.FloatingTask;
 import seedu.address.model.task.Task;
 import seedu.address.storage.Storage;
 
+
+
 /**
  * The main LogicManager of the app.
  */
 public class LogicManager extends ComponentManager implements Logic {
     private final Logger logger = LogsCenter.getLogger(LogicManager.class);
-
     private final Model model;
     private final Parser parser;
-
+    
+    //for undo
+    public static Stack<TaskBook> stateStack=new Stack<TaskBook>();//a stack of past TaskBook states
+    public static final Stack<Command> modifyingDataCommandHistory=new Stack<Command>();
+    
+    //for redo
+    public static Stack <TaskBook> undoneStates = new Stack<TaskBook>();
+    public static Stack<Command> undoneCommands=new Stack<Command>();
+    
```

Comment:
`static` fields that are not constants are a big no no. http://stackoverflow.com/questions/7026507/why-are-static-variables-considered-evil